src/main/java/org/opensearch/sdk/SDKHttpRequest.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.sdk;
+
+import org.opensearch.common.bytes.BytesReference;
+import org.opensearch.extensions.rest.ExtensionRestRequest;
+import org.opensearch.http.HttpRequest;
+import org.opensearch.http.HttpResponse;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.rest.RestStatus;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This class helps to get instance of RestRequest
+ */
+public class SDKHttpRequest implements HttpRequest {
+    private final RestRequest.Method method;
+    private final String uri;
+    private final BytesReference content;
+    private final Map<String, List<String>> headers;
+
+    /**
+    * Instantiates this class with a copy of request
+    *
+    * @param request The request
+    */
+    public SDKHttpRequest(ExtensionRestRequest request) {
+        this.method = request.method();
+        this.content = request.content();
+        this.headers = request.headers();
+
+        Path path = Paths.get(request.path());
```

Comment:
The REST path is not the same as a file path, so this isn't the right type to use here.

What I would like to do is change the `ExtensionRestRequest` itself to pass the `uri()` instead of the `path()`, and this would just be `this.uri = request.uri()`.

Reply:
> The REST path is not the same as a file path, so this isn't the right type to use here.
> 
> What I would like to do is change the `ExtensionRestRequest` itself to pass the `uri()` instead of the `path()`, and this would just be `this.uri = request.uri()`.

Should I just add uri to `ExtensionRestRequest`? Or I should also remove `path()`, `path` from `ExtensionRestRequest`

Reply:
> > The REST path is not the same as a file path, so this isn't the right type to use here.
> > What I would like to do is change the `ExtensionRestRequest` itself to pass the `uri()` instead of the `path()`, and this would just be `this.uri = request.uri()`.
> 
> Should I just add uri to `ExtensionRestRequest`? Or I should also remove `path()`, `path` from `ExtensionRestRequest`

You should add uri as a field and `uri()` as a method.

You should remove path as a field and `path()` as a method.

Reply:
(And you should also add a new `HttpRequest.HttpVersion` field and new method `protocolVersion()` to retrieve it.)